### DISCUSSION TITLE
Why does `concatMap((x) => x)` work the same as `concatMap((x) => from(x))`?

### DISCUSSION THREAD

**Discussion by ilyakonrad** [ReactiveX/rxjs#7233] [Q&A] [answered]
I am trying to understand the behavior of `concatMap()`.
Specifically the case when the projection function returns its argument as is. In my case it's an array.
I am trying to figure out **why `concatMap()` transforms emitted value (an array) into separate emissions of every item of the array**.

Here's the setup:

```
const source$ = of([1, 2, 3])

const resultConcat$ = source$.pipe(concatMap((x) => {
  console.log(x); // [1, 2, 3]
  return x
}))

resultConcat$.subscribe((value) => console.log(value))

// result:
// 1
// 2
// 3
```

It works the same when instead of doing `return x` I do `return from(x)`.
But then it makes sense, because `from()` is _supposed_ to emit items of the array separately.

What am I missing?

**Comment by demensky** [comment, marked as answer]
If you put a breakpoint inside the `project` function and then navigate up [you will see `from` there](https://github.com/ReactiveX/rxjs/blob/667e185435fbd6d83e92942779501bf960249948/src/internal/operators/mergeInternals.ts#L71).  
The example is shown in code from the `master` branch. But on the version that you use, the essence is about the same.

So yes, you don't need to use [`from`](https://rxjs.dev/api/index/function/from) in places where the type is specified as [`ObservableInput`](https://rxjs.dev/api/index/type-alias/ObservableInput) (which includes `ArrayLike<T>` and `Iterable<T>`).

**Comment by ilyakonrad** [comment]
Going up while debugging didn't get me to it, but I checked the source code and I got it.
I wonder why it's not clear from the documentation though.
Thank you for helping!

**Reply by jakovljevic-mladen** [reply]
There are known issues with the docs related to the `ObservableInput` parameters. We have a plan to update the documentation, we're sorry this has caused issues for you.
